src/main/java/com/traveltime/plugin/elasticsearch/query/TraveltimeWeight.java

```diff
@@ -68,9 +69,12 @@ public Scorer scorer(LeafReaderContext context) throws IOException {
       }
 
       val valueArray = new ArrayList<GeoPoint>();
+      val valueSet = new LongOpenHashSet();
 
       while (finalIterator.nextDoc() != DocIdSetIterator.NO_MORE_DOCS) {
-         valueArray.add(Util.decode(backing.nextValue()));
+         if(!valueSet.add(backing.nextValue())) {
```

Comment:
`nextValue` iterates over the values in a single document, which should be just a single coordinate. So this could technically skip all of the values.

For some reason the actual implementation always supplies the value on subsequent calls to `nextValue`, but I will fix this.